src/plugins/analysis/ipc/code/ipc_analyzer.py

```diff
@@ -0,0 +1,67 @@
+import json
+import tempfile
+from pathlib import Path
+
+from docker.types import Mount
+
+from analysis.PluginBase import AnalysisBasePlugin
+from helperFunctions.docker import run_docker_container
+from objects.file import FileObject
+
+DOCKER_IMAGE = 'ipc'
+
+class AnalysisPlugin(AnalysisBasePlugin):
+    '''
+    Inter-Process Communication Analysis
+    '''
+
+    # mandatory plugin attributes:
+    NAME = 'ipc_analyzer'  # name of the plugin (using snake case)
+    DESCRIPTION = 'Inter-Process Communication Analysis'# a short description of the plugin
+    VERSION = '0.1'  # the version of this plugin (should be updated each time the plugin is changed)
+    FILE = __file__  # used internally
+
+    # optional plugin attributes:
+    MIME_WHITELIST = ['application/x-executable', 'application/x-object', 'application/x-sharedlib']
+    DEPENDENCIES = ['file_type']  # list of plugin names that this plugin relies on (default: `[]`)
+    TIMEOUT = 600  # 10 minutes
+
+    def _run_ipc_analyzer_in_docker(self, file_object: FileObject) -> dict:
+        tmpDir = tempfile.TemporaryDirectory()
```

Comment:
> Variables should be named using "snake case" (see [PEP 8](https://peps.python.org/pep-0008/)) There are some more issues regarding PEP 8 compliance (only whitespace stuff):
> 
> > code/ipc_analyzer.py:13:1: E302 expected 2 blank lines, found 1
> > code/ipc_analyzer.py:20:57: E261 at least two spaces before inline comment
> > code/ipc_analyzer.py:60:1: W293 blank line contains whitespace
> > code/ipc_analyzer.py:67:27: W292 no newline at end of file
> 
> it should probably work to simply run [black](https://github.com/psf/black) on the files to fix those

Using snake case now for every variables and used black to fix styling issues.

Reply:
> What is more, `tempfile.TemporaryDirectory()` should be used with the "with" statement if possible. This not only handles the `.cleanup()` for us, it also makes sure that it is executed in case of an error (the way the code is currently structured, if an error happened between the creation of the temp dir and the cleanup, the cleanup would not be done).

Using `tempfile.TemporaryDirectory()` with the `with` statement now.